packages/sit-onyx/src/components/OnyxNavItem/OnyxNavItem.vue

```diff
@@ -0,0 +1,114 @@
+<script lang="ts" setup>
+import { computed } from "vue";
+import type { OnyxNavItemProps } from "./types";
+import OnyxListbox from "../OnyxListbox/OnyxListbox.vue";
+import type { ListboxOption } from "../OnyxListbox/types";
+
+const props = withDefaults(defineProps<OnyxNavItemProps>(), {
+  active: false,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the nav item is clicked (via click or keyboard).
+   */
+  navigate: [href: string];
+}>();
+
+defineSlots<{
+  /**
+   * An optional slot to show additional content behind the label (e.g. a `OnyxBadge`).
+   */
+  default?(): unknown;
+}>();
+
+const listboxOptions = computed<ListboxOption<string>[]>(() => {
+  return props.options?.map((opt) => ({ id: opt.href, label: opt.label })) ?? [];
+});
+</script>
+
+<template>
+  <li
+    role="menuitem"
+    tabindex="0"
+    :aria-label="props.label"
+    class="onyx-nav-item onyx-text"
+    :class="{ 'onyx-nav-item--active': props.active || props.options?.find((opt) => opt.active) }"
+    @click="props.href && emit('navigate', props.href)"
+    @keydown.enter="props.href && emit('navigate', props.href)"
+  >
+    <span>{{ props.label }}</span>
+    <slot></slot>
+  </li>
+  <OnyxListbox
+    v-if="listboxOptions.length > 0"
+    class="onyx-nav-item-listbox"
+    :label="'Options of ' + props.label"
+    :options="listboxOptions"
+    :model-value="props.options?.find((opt) => opt.active)?.href"
+    @update:model-value="$event && emit('navigate', $event)"
+  />
+</template>
+
+<style lang="scss">
+@use "../../styles/mixins/layers";
+
+.onyx-nav-item {
+  @include layers.component() {
+    display: inline-flex;
+    position: relative;
+    height: 2.5rem;
+    width: max-content;
+    padding: var(--onyx-spacing-2xs) var(--onyx-spacing-md);
+    justify-content: center;
+    align-items: center;
+    gap: var(--onyx-spacing-2xs);
+    flex-shrink: 0;
+    border-radius: var(--onyx-radius-sm);
+    background: var(--onyx-color-base-background-blank);
+    text-decoration: none;
+    font-family: var(--onyx-font-family);
+    color: var(--onyx-color-text-icons-neutral-medium);
+    cursor: pointer;
+
+    &:hover {
```

Comment:
```suggestion
    &:hover,
    &:focus-visible {{
```

The listbox should also open when tabbed via keyboard

Reply:
I implemented it, it opens when I focus the nav item, but then when I press 'tab' again it focuses the listbox for a short period of time and then the listbox disappears. Maybe you have an idea how to fix this?